File: Ex2/Tuan2/bai1/app/src/main/java/com/example/bai1/MainActivity.java
package com.example.bai1;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText editName;
    private EditText editAge;
    private TextView resultTextView;
    private Button checkButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        // Khởi tạo các thành phần giao diện
        editName = findViewById(R.id.editName);
        editAge = findViewById(R.id.editAge);
        resultTextView = findViewById(R.id.resultTextView);
        checkButton = findViewById(R.id.check);

        // Xử lý sự kiện khi nhấn Button
        checkButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                checkAge();
            }
        });

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    private void checkAge() {
        // Lấy giá trị từ EditText
        String name = editName.getText().toString().trim();
        String ageStr = editAge.getText().toString().trim();

        // Kiểm tra xem các trường có trống không
        if (name.isEmpty() || ageStr.isEmpty()) {
            Toast.makeText(this, "Vui lòng nhập họ và tên và tuổi!", Toast.LENGTH_SHORT).show();
            return;
        }

        // Chuyển đổi tuổi thành số nguyên
        int age;
        try {
            age = Integer.parseInt(ageStr);
        } catch (NumberFormatException e) {
            Toast.makeText(this, "Tuổi phải là số hợp lệ!", Toast.LENGTH_SHORT).show();
            return;
        }

        // Kiểm tra tuổi và hiển thị kết quả
        if (age > 65) {
            resultTextView.setText(name + " là người già.");
        } else if (6 <= age && age <= 65) {
            resultTextView.setText(name + " là người lớn.");
        }  else if (2 <= age && age < 6) {
            resultTextView.setText(name + " là trẻ em.");
        } else if (2 > age ) {
            resultTextView.setText(name + " là người lớn.");
        }else {
            Toast.makeText(this, "Tuổi phải lớn hơn 0!", Toast.LENGTH_SHORT).show();
        }
    }
}

File: bt2-th1/app/src/main/java/com/example/bt2th1/MainActivity.kt
package com.example.bt2th1

import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.bt2th1.databinding.ActivityMainBinding

private lateinit var binding: ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        addEvents()
    }

    private fun addEvents() {
        xulyCheck()
    }

    private fun xulyCheck() {
        binding.btnCheck.setOnClickListener{
            var ten = binding.edtName.text.toString()
            var age = binding.edtAge.text.toString().toInt()
            //kiem tra co nhap tuoi khong
            if (binding.edtAge.text.toString().isEmpty()){
                Toast.makeText(this, "Vui long nhap tuoi", Toast.LENGTH_SHORT).show()

            }
            if(age == null){
                Toast.makeText(this, "Tuoi phai la so", Toast.LENGTH_SHORT).show()
            }
            var category = when{
                age> 65 -> "Người già"
                age in 6..65 -> "Người lớn"
                age in 2..6 -> "Trẻ em"
                age > 0 -> "Em bé"
                else -> "Tuổi không hợp lệ"
            }
            binding.edtHienThi.setText("$ten thuộc nhóm: $category")
            binding.edtAge.setText("")
            binding.edtName.setText("")
        }
    }
}
